src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -512,8 +512,17 @@ internal override void Shutdown()
         internal override int Read(Span<byte> buffer)
         {
             ThrowIfDisposed();
-
-            return ReadAsync(buffer.ToArray()).AsTask().GetAwaiter().GetResult();
+            byte[] rentedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
```

Comment:
Should we leave a TODO if we want to rewrite that in the future?

Reply:
I'm not sure if we want to rewrite it. If needed that can be done as outcome of some investigation.